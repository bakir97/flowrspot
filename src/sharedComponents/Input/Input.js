import React from "react";
import styles from "./Input.module.scss";
const Input = ({ onChange, full, title, value, name, type }) => {
  return (
    <div className={full ? styles.container_full : styles.container}>
      <label className={styles.label}>{title}</label>
      <input
        onChange={onChange}
        className={styles.input}
        placeholder={title}
        value={value}
        name={name}
        type={type}
      />
    </div>
  );
};

export default Input;
