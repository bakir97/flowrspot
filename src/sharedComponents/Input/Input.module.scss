@mixin container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 50px;
  border-radius: 3px;
  border: solid 1px #dfe5ea;
  background-color: #f5f6f7;
  margin-bottom: 10px;
}
.container {
  @include container;
  width: 183px;
}
.container_full {
  @include container;
  width: 378px;
}
.input {
  padding: 0;
  font-family: inherit;
  color: #334144;
  font-size: 13px;
  margin-left: 15px;
  border: none;
  background-color: transparent;
  &:focus {
    outline: none;
  }
}
.label {
  margin-left: 15px;
  font-size: 10px;
  color: #949ea0;
  margin-bottom: 5px;
}
