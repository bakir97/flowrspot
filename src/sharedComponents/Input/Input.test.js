import React from "react";
import { shallow } from "enzyme";
import Input from "./Input";
describe("Input", () => {
  it("should render correctly", () => {
    const component = shallow(<Input />);
    expect(component).toMatchSnapshot();
  });
  it("should trigger onChange", () => {
    const mockFn = jest.fn();
    const component = shallow(<Input onChange={mockFn} />);
    component.find("input").simulate("change");
    expect(mockFn).toHaveBeenCalled();
  });
});
