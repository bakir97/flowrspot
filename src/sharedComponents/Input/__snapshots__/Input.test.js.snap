// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="container"
>
  <label
    className="label"
  />
  <input
    className="input"
  />
</div>
`;
